torch/CMakeLists.txt

```diff
@@ -598,11 +594,8 @@ if (BUILD_PYTHON)
   set(TORCH_PYTHON_LINK_FLAGS "")
 
   if (MSVC)
-    string(APPEND TORCH_PYTHON_LINK_FLAGS " /NODEFAULTLIB:LIBCMT.LIB")
+    string(APPEND TORCH_PYTHON_LINK_FLAGS " /NODEFAULTLIB:LIBCMT.LIB $<$<NOT:$<CONFIG:Release>>:/DEBUG:FULL>")
```

Comment:
The build failure indicates that this clause is not correctly expanded as you expected. Is this only supported in newer version of CMake?

Reply:
Well, from this [issue](https://gitlab.kitware.com/cmake/cmake/issues/17611), looks like the generator expressions are not supported to be used with `set_target_properties`.

Reply:
Would you please change it to use `LINK_OPTIONS`, which supports the generator expressions?

Reply:
Only really care about that first commit. I'll just remove that change.
Thanks.